File: secure-data-manager/secure-data-manager-backend/src/main/java/ch/post/it/evoting/securedatamanager/shared/workflow/WorkflowState.java
/*
 * (c) Copyright 2025 Swiss Post Ltd.
 */
package ch.post.it.evoting.securedatamanager.shared.workflow;

import java.time.Instant;

public record WorkflowState(WorkflowStep step, Instant startTimestamp, Instant endTimestamp, WorkflowStatus status, String contextId,
							WorkflowExceptionCode exceptionCode, boolean optional) {
}

File: secure-data-manager/secure-data-manager-frontend/libs/shared/util-types/src/lib/workflow-state.ts
/*
 * (c) Copyright 2025 Swiss Post Ltd.
 */
import {WorkflowStatus} from './workflow-status';
import {WorkflowStep} from './workflow-step';
import {WorkflowExceptionCode} from './workflow-exception-code';

export interface WorkflowState {
	step: WorkflowStep;
	contextId?: string;
	status: WorkflowStatus;
	startTimestamp: string | null;
	endTimestamp: string | null;
	exceptionCode?: WorkflowExceptionCode;
	optional: boolean;
}
